### PR TITLE
Add: utils for getting hexseed from mnemonic

### PR DESCRIPTION
**Pull request by Prajjawalk**

### CODE REVIEW

File: src/utils/mnemonic.js [hunk 1]
```diff
@@ -29,4 +29,64 @@ function SeedBinToMnemonic(input) {
   return binToMnemonic(input);
 }
 
-module.exports = { SeedBinToMnemonic };
+function mnemonicToBin(mnemonic) {
+  mnemonicWords = mnemonic.split(' ')
+  wordCount = mnemonicWords.length
+  if (wordCount % 2 != 0) {
```
Comment (jplomas) [inline src/utils/mnemonic.js:35]: !== is preferred to != due to type checking being included with !==

Example:

(0 ==  '0') // true
(0 === '0') // false

('' ==  0 ) // true, the string will implicitly be converted to an integer
('' === 0 ) // false, no implicit cast is being made

File: src/utils/mnemonic.js [hunk 2]
```diff
@@ -29,4 +29,64 @@ function SeedBinToMnemonic(input) {
...
+
+  result = new Uint8Array(wordCount * 15 / 10)
+
+  let current = 0
+  let buffering = 0
+  let resultIndex = 0
+
+  mnemonicWords.map((w) => {
+    value = wordLookup[w]
+    if (value == undefined || value == null) {
```
Comment (jplomas) [inline src/utils/mnemonic.js:52]: === is preferred here

**Review by jplomas — CHANGES_REQUESTED** [review]
Code should be run through eslint as there are some style enforcements setup that will pick up some areas of fragility including the !=/!== and ==/=== issue as commented on it review.

npm run lint-check
will run the code through eslint

npm run lint
will do the same but also make any fixes automatically

File: src/utils/mnemonic.js [hunk 3]
```diff
@@ -29,4 +29,64 @@ function SeedBinToMnemonic(input) {
...
+    result[resultIndex] = current & 0xFF
+    resultIndex++
+  }
+  return result
+}
+
+function MnemonicToSeedBin(mnemonic) {
+  let output = mnemonicToBin(mnemonic)
+
+  if (output.length != 48) {
```
[the remark this answers is not in the document]
Comment (Prajjawalk) [reply src/utils/mnemonic.js]: sure, fixed linting errors

File: src/utils/mnemonic.js [hunk 4]
```diff
@@ -29,4 +29,65 @@ function SeedBinToMnemonic(input) {
   return binToMnemonic(input);
 }
 
-module.exports = { SeedBinToMnemonic };
+function mnemonicToBin(mnemonic) {
+  const mnemonicWords = mnemonic.split(' ');
+  const wordCount = mnemonicWords.length;
+  if (wordCount % 2 !== 0) {
+    throw new Error('word count must be even');
```
Comment (jplomas) [inline src/utils/mnemonic.js:36]: This conditional needs a unit test

File: src/utils/mnemonic.js [hunk 5]
```diff
@@ -29,4 +29,65 @@ function SeedBinToMnemonic(input) {
...
+  const result = new Uint8Array((wordCount * 15) / 10);
+
+  let current = 0;
+  let buffering = 0;
+  let resultIndex = 0;
+
+  mnemonicWords.map((w) => {
+    const value = wordLookup[w];
+    if (value === undefined || value === null) {
+      throw new Error('invalid word in mnemonic');
```
Comment (jplomas) [inline src/utils/mnemonic.js:54]: This conditional needs a unit test

File: src/utils/mnemonic.js [hunk 6]
```diff
@@ -29,4 +29,65 @@ function SeedBinToMnemonic(input) {
...
+    resultIndex++;
+  }
+  return result;
+}
+
+function MnemonicToSeedBin(mnemonic) {
+  const output = mnemonicToBin(mnemonic);
+
+  if (output.length !== 48) {
+    throw new Error('unexpected MnemonicToSeedBin output size');
```
Comment (jplomas) [inline src/utils/mnemonic.js:85]: This conditional needs a unit test

**Review by jplomas — CHANGES_REQUESTED** [review]
A few conditionals need unit tests as indicated

File: src/utils/mnemonic.js (hunk 6 above)
Comment (Prajjawalk) [reply src/utils/mnemonic.js:85]: yep, added unit tests for the above conditionals

**Review by jplomas — APPROVED** [review]

**Pull request merged**